connect/src/com/telenor/connect/AbstractSdkProfile.java

```diff
@@ -166,6 +166,10 @@ protected void previewParameters(ParametersHolder parameters) {
                         && isEmpty(wellKnownConfig.getNetworkAuthenticationTargetUrls()))) {
                     parameters.add("login_hint", String.format("MCCMNC:%s", mccMnc));
         }
+
+        if (!ConnectSdk.isCellularDataNetworkConnected()) {
```

Comment:
No, it's fine. I mean this has been already considered in 

Check comments under 
https://confluence.skunk-works.no/confluence/display/Scale/Facilitating+Header+Enrichment+over+WiFi+in+MobileConnect+%28and+Connect+ID%29+SDK
and look for " We rely on the requestNetwork callbacks" for possible callbacks/adapters states

But I'm afraid we can't rely on that here, so maybe something like what you are doing will be necessary.